heartbeat/scheduler/timerqueue/queue.go

```diff
@@ -103,16 +114,22 @@ func (tq *TimerQueue) Start() {
 func (tq *TimerQueue) pushInternal(tt *timerTask) {
 	heap.Push(&tq.th, tt)
 
-	if tq.nextRunAt == nil || tq.nextRunAt.After(tt.runAt) {
-		// Stop and drain the timer prior to reset per https://golang.org/pkg/time/#Timer.Reset
-		// Only drain if nextRunAt is set, otherwise the timer channel has already been stopped the
-		// channel is empty (and thus would block)
-		if tq.nextRunAt != nil && !tq.timer.Stop() {
-			<-tq.timer.C
-		}
-		tq.timer.Reset(tt.runAt.Sub(time.Now()))
-
+	if tq.nextRunAt == nil {
+		tq.timer.Reset(time.Until(tt.runAt))
```

Comment:
Is this another instance where it is resetting the timer after stopping it (timer fires, heap is emptied, timer is stopped, new task comes in)? I guess this and the other `Reset` should be replaced too.